pkg/plugin/plugin.go

```diff
@@ -12,21 +14,38 @@ import (
 	"github.com/openshift/openshift-azure/pkg/validate"
 )
 
-type Plugin struct{}
+type plugin struct {
+	log *logrus.Entry
+}
+
+var _ api.Plugin = &plugin{}
 
-var _ api.Plugin = &Plugin{}
+func NewPlugin(log *logrus.Entry) api.Plugin {
+	// always have a logger
+	if log == nil {
+		logger := logrus.New()
+		log = logrus.NewEntry(logger)
+	}
+
+	return &plugin{
+		log: log,
+	}
+}
 
-func (p *Plugin) ValidateInternal(new, old *acsapi.ContainerService) []error {
+func (p *plugin) ValidateInternal(new, old *acsapi.ContainerService) []error {
+	p.log.Info("validating internal data models")
```

Comment:
I dont like syntax like this, but not sure if anything could be done here :/

Reply:
yeah, that's why I was hoping we could just configure the global level logger but I didn't see a good way to do that when you want something like custom `WithFields` that was mentioned.  The other option I could think of would be to wrap logging our own package like

```go
package log

struct LogWrapper { log *logrus.Entry }

var logger *LogWrapper

//method to create logger once

//static wrapper methods for log.*
```

Then you could import our logging package and it would use the instantiated `LogWrapper`.